messaging/broker.go

```diff
@@ -619,9 +631,13 @@ const DefaultMaxSegmentSize = 10 * 1024 * 1024 // 10MB
 type Topic struct {
 	mu    sync.Mutex
 	id    uint64 // unique identifier
-	index uint64 // highest index replicated
```

Comment:
The field was accidentally being overloaded before and it stored the last index written to the topic at first and then used for heartbeats. The heartbeat tracking was moved to the `indexByURL` field.